ecoscope/base/trajectory.py

```diff
@@ -0,0 +1,402 @@
+import itertools
+import warnings
+from copy import deepcopy
+
+import geopandas as gpd  # type: ignore[import-untyped]
+import numpy as np
+import pandas as pd
+import shapely
+from pyproj import Geod
+
+from ecoscope.base._dataclasses import (
+    ProximityProfile,
+    TrajSegFilter,
+)
+from ecoscope.base.straightrack import StraightTrackProperties
+from ecoscope.base import EcoDataFrame, Relocations
+
+
+def get_displacement(gdf: gpd.GeoDataFrame):
+    """
+    Get displacement in meters between first and final fixes.
+    """
+    if not gdf["segment_start"].is_monotonic_increasing:
+        gdf = gdf.sort_values("segment_start")
+    start = gdf.geometry.iloc[0].coords[0]
+    end = gdf.geometry.iloc[-1].coords[1]
+    return Geod(ellps="WGS84").inv(start[0], start[1], end[0], end[1])[2]
+
+
+def get_tortuosity(gdf: gpd.GeoDataFrame):
+    """
+    Get tortuosity for dataframe defined as distance traveled divided by displacement between first and final
+    points.
+    """
+
+    return gdf["dist_meters"].sum() / get_displacement(gdf)
+
+
+class Trajectory(EcoDataFrame):
+    """
+    A trajectory represents a time-ordered collection of segments.
+    Currently only straight track segments exist.
+    It is based on an underlying relocs object that is the point representation
+    """
+
+    @classmethod
+    def from_relocations(cls, relocs: Relocations, *args, **kwargs):
```

Comment:
```suggestion
    def from_relocations(cls, relocs: Relocations, copy: bool = True):
```
Same comment as above. I think we should try to avoid open-ended `**kwargs` wherever possible. I realize that this is just copy-and-paste from the prior implementation, so not a conscious choice here, but it felt too glaring, and the fix just too easy, to not propose.